src/pages/home/report/ReportActionsView.js

```diff
@@ -109,9 +110,12 @@ class ReportActionsView extends React.Component {
         if (this.keyboardEvent) {
             this.keyboardEvent.remove();
         }
+
         if (this.visibilityChangeEvent) {
             this.visibilityChangeEvent.remove();
         }
+
+        _.each(this.timers, timer => clearTimeout(timer));
```

Comment:
These timers would fire after logging out when switching to an unread report and then signing out. Any timer that makes a network request should be canceled if we can't make that request anymore.

Reply:
Good call

Reply:
There is also a timer for fetching the personal details every 30 minutes. Do we need to find a way to cancel that one as well? It's a little harder since it's not attached to any view and only handled in the `PersonalDetails.js` action.

Reply:
We shouldn't need to cancel that as it will just run for the entire life of the executing JS. It will never be called more than once so it cannot cause memory leaks and also cannot execute after the window has been closed.

Reply:
Oh wait a second.... I see the ones you are talking about.

Reply:
I think probably we should just clear whatever timer exists before we set them again. But that seems like less of a priority. Worst case scenario they will just be called more frequently.

Reply:
Gonna leave this for another time.